var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -486,11 +486,11 @@ def edit(self, spec, prefix):
 
             # Currently AOCC support only static libraries of ELPA
             if '%aocc' in spec:
-                libs.append(join_path(elpa.libs.directories[0],
+                libs.append(join_path(elpa.prefix.lib,
                             ('libelpa{elpa_suffix}.a'
                                 .format(elpa_suffix=elpa_suffix))))
             else:
-                libs.append(join_path(elpa.libs.directories[0],
+                libs.append(join_path(elpa.prefix.lib,
```

Comment:
NACK. `elpa.prefix` is a dir, hence `elpa.prefix.lib` is the content of `elpa.prefix` with `/lib` appended. But `elpa` sometimes uses `lib` and sometimes `lib64`. See for example on an openSUSE:

```
opt/spack/linux-opensuse_leap15-zen/gcc-11.2.0/elpa-2021.05.002_bugfix-2tpqafsmqpplsgx6c3ueexresram2qus/lib64/libelpa_openmp.a
opt/spack/linux-opensuse_leap15-zen/gcc-11.2.0/elpa-2021.05.002_bugfix-2tpqafsmqpplsgx6c3ueexresram2qus/lib64/libelpa_openmp.so.15
opt/spack/linux-opensuse_leap15-zen/gcc-11.2.0/elpa-2021.05.002_bugfix-2tpqafsmqpplsgx6c3ueexresram2qus/lib64/libelpa_openmp.so.15.1.2
```

Reply:
Thanks for the review, I am okay if it was taking the `/lib` or `/lib64` based on what is available in `elpa.prefix`, but it is not picking the `/lib `for `Linux and AOCC` case, thats why we have undone he change.

Reply:
Then you have to find another solution as this will break the build on machines I maintain. Since the `elpa` package definitely has a `libs` method my guess would be that you were using an outdated version of the elpa package, but maybe it was something else.

Reply:
we are using `ELPA 2017` for building `CP2K v6.1`.
Will revert my changes for `else clause` but for `%aocc` it is required to be passed that way.